### PR TITLE
Rewrite test_reloc, test_scptid, test_cfg_tree based on criterion

### PR DESCRIPTION
**Pull request by norberttak**

### CODE REVIEW

File: lib/tests/test_rcptid.c [hunk 1]
```diff
@@ -36,6 +37,7 @@ PersistState *state;
 void
 setup_persist_id_test(void)
 {
+  app_startup();
```
Comment (Kokan) [inline lib/tests/test_rcptid.c]: The `teardown_persist_id_test` should call the `app_shutdown`.

**Comment by MrAnno** [comment]
Please do not forget to mark these tests in CMakeLists.txt (CRITERION).

File: lib/tests/test_rcptid.c (hunk 1 above)
Comment (norberttak) [reply lib/tests/test_rcptid.c]: add missing app_shutdown: done

**Comment by norberttak** [comment]
I added the missing CMakeLists.txt as well

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/tests/test_cfg_tree.c [hunk 2]
```diff
@@ -92,135 +92,112 @@ create_and_attach_almighty_pipe (CfgTree *tree, gboolean pipe_value)
...
+  gboolean tree_stop_expected;
+  gboolean was_init_called;
+  gboolean was_deinit_called;
+} PipeParameter;
+
+ParameterizedTestParameters(cfg_tree, test_pipe_init)
+{
+  static PipeParameter test_data_list[] =
+  {
+    {TRUE, TRUE, TRUE, TRUE, TRUE},
```
Comment (furiel) [inline lib/tests/test_cfg_tree.c]: I suggest using initializers here: { .always_pipe_value = TRUE, .tree_start_expected = TRUE, [etc ...]}.

**Comment by furiel** [comment]
Is there any reason we need to call each testcases in test_rcptid sequencially? It would be great if they could be separated into different criterion Test macros. Certainly, a small adjustment would be necessary: the persist state file must be different for each tests, because the tests are running parallel, overwriting each-others persist file.

File: lib/tests/test_cfg_tree.c (hunk 2 above)
Comment (norberttak) [reply lib/tests/test_cfg_tree.c]: Done

**Comment by norberttak** [comment]
test_rcptid: I did the modifications. Now test cases are able to run in paralel.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**